File: src/led/ColorHSV.java
package led;

public class ColorHSV {
	public int h;
	public float s;
	public float v;
	
	public ColorHSV(int _h, float _s, float _v) {
		h = _h;
		s = _s;
		v = _v;
	}
	
	public ColorRGB ToRGB() {
		int v255 = (int) (v * 255.0f);
		if(s == 0.0f) { return new ColorRGB(v255, v255, v255); }
		
		int hueInterval = (int) Math.floor(h / 60.0f);
        float f = (h / 60.0f) - (float) hueInterval;
        
        int p = (int) ((v * (1 - s)) * 255.0f);
        int q = (int) ((v * (1 - s * f)) * 255.0f);
        int t = (int) ((v * (1 - s * (1 - f))) * 255.0f);
        
        ColorRGB result = 
        		hueInterval == 0 ||
        		hueInterval == 6 ? 	new ColorRGB(v255, t, p) :
    			hueInterval == 1 ? 	new ColorRGB(q, v255, p) :
				hueInterval == 2 ? 	new ColorRGB(p, v255, t) :
				hueInterval == 3 ? 	new ColorRGB(p, q, v255) :
				hueInterval == 4 ? 	new ColorRGB(t, p, v255) :
				hueInterval == 5 ? 	new ColorRGB(v255, p, q) : 
								   	new ColorRGB(0, 0, 0);
        
        return result;
	}
}

File: src/main/java/led/ColorHSV.kt
package led

class ColorHSV(var h: Int, var s: Float, var v: Float) {
    fun ToRGB(): ColorRGB {
        val v255 = (v * 255.0f).toInt()
        if (s == 0.0f) {
            return ColorRGB(v255, v255, v255)
        }
        val hueInterval = Math.floor((h / 60.0f).toDouble()).toInt()
        val f = h / 60.0f - hueInterval.toFloat()
        val p = (v * (1 - s) * 255.0f).toInt()
        val q = (v * (1 - s * f) * 255.0f).toInt()
        val t = (v * (1 - s * (1 - f)) * 255.0f).toInt()
        return if (hueInterval == 0 ||
            hueInterval == 6
        ) ColorRGB(v255, t, p) else if (hueInterval == 1) ColorRGB(q, v255, p) else if (hueInterval == 2) ColorRGB(
            p,
            v255,
            t
        ) else if (hueInterval == 3) ColorRGB(p, q, v255) else if (hueInterval == 4) ColorRGB(
            t,
            p,
            v255
        ) else if (hueInterval == 5) ColorRGB(v255, p, q) else ColorRGB(0, 0, 0)
    }
}